### ISSUE TITLE
apk add aws-cli using edge repository returns error

### ISSUE THREAD

**Issue by john1001101** [alpinelinux/docker-alpine#319]
Docker images that use edge repository http://dl-3.alpinelinux.org/alpine/edge/community/ when installing aws-cli returns the following error:

```
Sending build context to Docker daemon  2.698MB
Step 1/2 : FROM alpine:3
 ---> 9ed4aefc74f6
Step 2/2 : RUN apk add --no-cache ca-certificates     --repository http://dl-3.alpinelinux.org/alpine/edge/community/     aws-cli
 ---> Running in 1bb095f0a571
fetch http://dl-3.alpinelinux.org/alpine/edge/community/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/x86_64/APKINDEX.tar.gz
ERROR: unable to select packages:
  python3-3.10.11-r0:
    breaks: py3-colorama-0.4.6-r3[python3~3.11]
            py3-cryptography-40.0.2-r1[python3~3.11]
    satisfies: py3-wcwidth-0.2.5-r2[python3]
               py3-urllib3-1.26.12-r0[python3]
               py3-docutils-0.19-r1[python3]
               py3-prompt_toolkit-3.0.38-r0[python3]
               py3-colorama-0.4.6-r3[python3]
               py3-dateutil-2.8.2-r2[python3]
               aws-cli-2.11.13-r0[python3]
               py3-cryptography-40.0.2-r1[python3]
               py3-jmespath-1.0.1-r0[python3]
               py3-ruamel.yaml.clib-0.2.7-r1[python3]
               py3-cparser-2.21-r0[python3]
               py3-distro-1.8.0-r1[python3]
               py3-certifi-2022.12.7-r0[python3]
               py3-six-1.16.0-r3[python3]
The command '/bin/sh -c apk add --no-cache ca-certificates     --repository http://dl-3.alpinelinux.org/alpine/edge/community/     aws-cli' returned a non-zero code: 17
```

**Steps to reproduce**
Create Dockerfile with following:

```
FROM alpine:3

RUN apk add --no-cache ca-certificates \
    --repository http://dl-3.alpinelinux.org/alpine/edge/community/ \
    aws-cli
```

`docker build .`

**Comment by ptrcnull** [comment]
mixing edge and stable repositories is not supported. the issue you see is happening, because `py3-colorama` and `py3-cryptography` depend on python 3.11, which is not available in the 3.17 main repository

**Comment by john1001101** [comment]
Understood, thanks for clarifying this to me. I was curious about this because we had a service we we're dependent on that had this issue. We ended up forking our own and removing the edge repository. 

Thanks for the help
